processor/src/main/java/org/robolectric/annotation/processing/generator/ShadowProviderGenerator.java

```diff
@@ -163,11 +163,11 @@ void generate(PrintWriter writer) {
 
     writer.println("  @Override");
     writer.println("  public String[] getProvidedPackageNames() {");
-    writer.println("    return new String[] {");
+    writer.println("    return new LinkedHashSet<>(Arrays.asList(");
```

Comment:
Explanation:
for the support lib, we'd have:
```
        "android.support.v4.content",
        "android.support.v4.content",
        "android.support.v4.media",
        "android.support.v4.widget",
        "android.support.v4.widget"
```

for androidx, we'd have:
```
        "android.support.v4.media",
        "androidx.drawerlayout.widget",
        "androidx.loader.content",
        "androidx.localbroadcastmanager.content",
        "androidx.swiperefreshlayout.widget"
```

We can't detect the duplicates at the time we create this list (in `RobolectricModel`), so if we want to avoid returning duplicates in this method, we have to do it at runtime, and put them into a `Set`, here.